checker/src/main/java/org/checkerframework/checker/nullness/NullnessAnnotatedTypeFactory.java

```diff
@@ -474,6 +470,9 @@ public Void visitNewClass(NewClassTree node, AnnotatedTypeMirror type) {
 
         @Override
         public Void visitNewArray(NewArrayTree node, AnnotatedTypeMirror type) {
+            // The result of newly allocated structures is always non-null.
+            type.replaceAnnotation(NONNULL);
```

Comment:
On master I tried

```java
  void foo(Object nn) {
    nn = new int @Nullable [5];
  }
```

and this gives an assignment error.
With this change it looks like the code would be quietly accepted.

I see two options:
- only add NONNULL if nothing is present yet, which is the old behavior; this isn't ideal, because a new array always is non-null. So as a follow-up issue we can validate those type better.
- raise an error here if there already is a qualifier other than NONNULL; this isn't ideal, as we don't like raising errors in the type factory.